pkgs/stdenv/linux/default.nix

```diff
@@ -328,12 +328,12 @@ in
     assert isBuiltByBootstrapFilesCompiler prevStage.patchelf;
     stageFun prevStage {
       name = "bootstrap-stage-xgcc";
-      overrides = final: prev: {
```

Comment:
I thought some people were trying to move away from `self` `super`?

Reply:
See my commit message:

> As of late, `final: prev: …` for overlays has become more prevalent in
> newer code. This is also exhibited in some code (presumably added
> recently) in stdenv. This change is not about any merits or demerits of
> any naming convention, but rather aims to make the nomenclature in
> stdenv bootstrapping consistent to lessen confusion.
>
> I've chosen to stick to `self: super: …` convention because:
>
> 1. It is more common in the code as it stands.
>
> 2. Using `final: prev: …` makes the code more confusing, als it causes
>    `prev` to be in scope alongside `prevStage`. `prevStage` actually
>    bears no relation to `prev` even though their naming suggests it,
>    making it easy to confuse them (the former is the (final) package set
>    of the previous stage while the latter is just the `prev`/`super` of
>    the overlay “chaining” on a completely fresh package set, i.e. `prev`
   doesn't even relate to the previous stage's `overrides` argument).

I'd be open to change it at a later date, though.

Reply:
OK that's fair enough reasoning for me. If the bootstrap is finally made non-bad (not so many stages, uses cross stuff), we can revisit these things.